### DISCUSSION TITLE
How often does the action actually run?

### DISCUSSION THREAD

**Discussion by ffisc** [skills/.github#149] [Q&A]
Step 5 mentions "**Your new action** will run any time a new commit is **created** or **pushed** to the remote repository."

But "my new action" (i.e. `welcome.yml`) is set up as

```
on:
  pull_request:
    types: [opened]
```

Which leads me to think it will run only when a PR is opened, not every time a commit is pushed.
Am I missing something obvious here?

**Comment by Mahiuha** [comment]
Yes, you are correct that the action defined in your **welcome.yml** file will only run when a pull request is opened, not every time a commit is pushed. The step you mentioned is likely referencing a different event that can trigger an action, such as a push event.
The **on** section of the YAML file determines when the action is executed, and in your case, it's set to trigger when a pull request is opened. So, the action will only run when a pull request is opened, not when a commit is pushed.
It is important to note that you can have multiple triggers defined in the on section such as pull_request, push, etc. So you can set different triggers for different actions.
